transports/tcp/src/lib.rs

```diff
@@ -605,23 +607,12 @@ pub enum TcpListenerEvent<S> {
     Error(io::Error),
 }
 
-enum IfWatch<TIfWatcher> {
-    Pending(BoxFuture<'static, io::Result<TIfWatcher>>),
-    Ready(TIfWatcher),
-}
-
 /// The listening addresses of a [`TcpListenStream`].
-enum InAddr<TIfWatcher> {
+enum InAddr {
     /// The stream accepts connections on a single interface.
-    One {
-        addr: IpAddr,
-        out: Option<Multiaddr>,
-    },
+    One(Option<Multiaddr>),
```

Comment:
The use of `Option` here is pretty subtle. Would it make sense to introduce a 3rd variant that represents the state of already having reported the multiaddr once? This would also allow us to use `mem::replace` so we don't need to `take` the `Multiaddr` from the `Option`.

Reply:
Removed as suggested in https://github.com/libp2p/rust-libp2p/pull/2813#issuecomment-1220538958, hence this is obsolete.